### PR TITLE
Give all jobs a variant label that need one

### PR DESCRIPTION
**Pull request by smarterclayton**
Labels should be used to carry categorization info to jobs, and variant
is an excellent one (like artifacts).  Allows validation in https://github.com/openshift/release/pull/2102 to correctly identify variants.

### CODE REVIEW

File: cmd/ci-operator-prowgen/main_test.go [hunk 1]
```diff
@@ -388,10 +397,13 @@ func TestGeneratePostSubmitForTest(t *testing.T) {
 			treatBranchesAsExplicit: true,
 
 			expected: &prowconfig.Postsubmit{
-				Agent:    "kubernetes",
+				Agent: "kubernetes",
+				Labels: map[string]string{
+					"artifacts":                        "images",
+					"ci-operator.openshift.io/variant": "",
```
Comment (stevekuznetsov) [inline cmd/ci-operator-prowgen/main_test.go]: Add the label values as a constant please

File: test/integration/data/output/jobs/super/duper/super-duper-release-3.11-postsubmits.yaml [hunk 2]
```diff
@@ -6,6 +6,7 @@ postsubmits:
     decorate: true
     labels:
       artifacts: images
+      ci-operator.openshift.io/variant: ""
```
Comment (stevekuznetsov) [inline test/integration/data/output/jobs/super/duper/super-duper-release-3.11-postsubmits.yaml]: Can we treat a missing label the same way as empty? I don't want to see this add a linearly scaling number of lines to our config ...

Comment (smarterclayton) [reply test/integration/data/output/jobs/super/duper/super-duper-release-3.11-postsubmits.yaml]: we could, but the empty value is a lot nicer for people to integrate with.  it's up to you.  I think about all the places where someone wants to find and replace etc.

**Comment by petr-muller** [comment]
/retest
how about now?

**Comment by petr-muller** [comment]
/retest
and now?

**Comment by petr-muller** [comment]
/retest
pretty please?

File: test/integration/data/output/jobs/super/duper/super-duper-release-3.11-postsubmits.yaml (hunk 2 above)
Comment (stevekuznetsov) [reply test/integration/data/output/jobs/super/duper/super-duper-release-3.11-postsubmits.yaml]: I don't see how it's easier to integrate -- if you're opting in you're adding a line anyway right?

Comment (smarterclayton) [reply test/integration/data/output/jobs/super/duper/super-duper-release-3.11-postsubmits.yaml]: I can remove it, makes the change smaller

**Comment by smarterclayton** [comment]
Ok, comments addressed.  Label is optional, has a constant.

**Review by petr-muller — APPROVED** [review]

**Pull request merged**